File: research/stellar/examples/javascript/account/activate-account.js
const StellarSdk = require('stellar-sdk')
const server = new StellarSdk.Server('https://horizon-testnet.stellar.org')

var argv = require('yargs')
    .usage('Usage: $0 --sourceKey [string] --destinationAddress [string]')
    .demandOption(['sourceKey', 'destinationAddress'])
    .argv;

async function fundAccount (sourceKey, destination) {
  const sourceKeypair = StellarSdk.Keypair.fromSecret(sourceKey)
  const account = await server.loadAccount(sourceKeypair.publicKey());

  const fee = await server.fetchBaseFee();

  const createTransaction = new StellarSdk.TransactionBuilder(account, {
    fee,
    networkPassphrase: StellarSdk.Networks.TESTNET
  })
  .addOperation(StellarSdk.Operation.createAccount({
    destination,
    startingBalance: '100'
  }))
  .setTimeout(30)
  .build()

  createTransaction.sign(sourceKeypair)

  console.log(createTransaction.toEnvelope().toXDR('base64'));

  try {
    const transactionResult = await server.submitTransaction(createTransaction);
    console.log(JSON.stringify(transactionResult, null, 2));
    console.log('\nSuccess! View the transaction at: ');
    console.log(transactionResult._links.transaction.href);
  } catch (e) {
    console.log('An error has occured:');
    console.log(e);
  }
}

fundAccount(argv.sourceKey, argv.destinationAddress)

File: research/stellar/examples/python/account/activate-account.py
# pylint: disable=no-value-for-parameter
from stellar_sdk import TransactionBuilder, Server, Network, Keypair
from stellar_sdk.exceptions import BadRequestError
import click

@click.command()
@click.option('--sourcekey', help='Secret key of the source', required=True)
@click.option('--destinationaddress', help='Destination address of the account to activate', required=True)

def activate_account(sourcekey, destinationaddress):
  server = Server(horizon_url="https://horizon-testnet.stellar.org")
  source = Keypair.from_secret(sourcekey)

  source_account = server.load_account(account_id=source.public_key)
  transaction = TransactionBuilder(
      source_account=source_account,
      network_passphrase=Network.TESTNET_NETWORK_PASSPHRASE,
      base_fee=100) \
      .append_create_account_op(destination=destinationaddress, starting_balance="12.25") \
      .build()
  transaction.sign(source)
  try:
    response = server.submit_transaction(transaction)
    print("Transaction hash: {}".format(response["hash"]))
    print(response)
  except BadRequestError as e:
    print(e)

if __name__ == '__main__':
  activate_account()